### PR TITLE
Refactor Job RPC Services

### PR DESCRIPTION
**Pull request by tgianos**

### CODE REVIEW

File: genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverter.java [hunk 1]
```diff
@@ -88,62 +93,86 @@ public static GetJobSpecificationRequest toProtoGetJobSpecificationRequest(final
         return GetJobSpecificationRequest.newBuilder().setId(id).build();
     }
 
+//    /**
```
Comment (mprimi) [inline genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverter.java]: Meant to add all this commented out code?

File: genie-common-internal/src/test/groovy/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverterSpec.groovy [hunk 2]
```diff
@@ -164,24 +196,39 @@ class JobSpecificationServiceConverterSpec extends Specification {
 
     def environmentVariables = ImmutableMap.of(UUID.randomUUID().toString(), UUID.randomUUID().toString())
 
-    def "Can convert JobRequest to ResolveJobSpecificationRequest and vice versa"() {
-        def jobRequest = createJobRequest()
+//    def "Can convert JobRequest to ResolveJobSpecificationRequest and vice versa"() {
```
Comment (mprimi) [inline genie-common-internal/src/test/groovy/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverterSpec.groovy]: More here

File: genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobServiceImpl.java [hunk 3]
```diff
@@ -34,29 +34,29 @@
...
 @Slf4j
-public class GRpcJobSpecificationServiceImpl extends JobSpecificationServiceGrpc.JobSpecificationServiceImplBase {
+public class GRpcJobServiceImpl extends JobServiceGrpc.JobServiceImplBase {
 
     private final JobSpecificationService jobSpecificationService;
 
     /**
      * Constructor.
-     *
+     *GRpcJobServiceImplSpec
```
Comment (mprimi) [inline genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobServiceImpl.java]: Stray line?

File: genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverter.java (hunk 1 above)
Comment (tgianos) [reply genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobSpecificationServiceConverter.java]: Yeah I left a bunch of this stuff in due to expecting to need it once the reservation of ID code is written it's going to be moved there. in there interest of getting this done quickly I left it like this

File: genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobServiceImpl.java (hunk 3 above)
Comment (tgianos) [reply genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobServiceImpl.java]: yeah probably something while i was debugging and got in here due to a pasting in the wrong spot. I'll remove it

**Pull request merged**